False positive in nimble_operator rule

### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug


##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint --path file.swift --enable-all-rules --no-cache | grep nimble
Loading configuration from '.swiftlint.yml'
Linting Swift files at paths file.swift
Linting 'file.swift' (1/1)
Done linting! Found 2 violations, 0 serious in 1 file.
file.swift:2:2: warning: Nimble Operator Violation: Prefer Nimble operator overloads over free matcher functions. (nimble_operator)
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.28.2
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Homebrew
* Paste your configuration file:

```yml
# insert yaml contents here
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.

No

* Which Xcode version are you using (check `xcode-select -p`)?

10.1
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
// This triggers a violation:
foo.method {
    expect(value).to(equal(expectedValue), description: "Failed")
    return Bar(value: ())
}
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -17,7 +17,9 @@
 
 #### Bug Fixes
 
-* None.
+* Fix false positive in `nimble_operator` rule.  
+  [Marcelo Fabri](https://github.com/marcelofabri)
+  [#2489](https://github.com/realm/SwiftLint/issues/2489)
 
 ## 0.29.0: A Laundry List of Changes
 

--- a/Rules.md
+++ b/Rules.md
@@ -11925,6 +11925,13 @@ expect(actual).to(haveCount(expected))
 
 ```
 
+```swift
+foo.method {
+    expect(value).to(equal(expectedValue), description: "Failed")
+    return Bar(value: ())
+}
+```
+
 </details>
 <details>
 <summary>Triggering Examples</summary>
@@ -11941,6 +11948,11 @@ expect(actual).to(haveCount(expected))
 
 ```swift
 ↓expect(10).to(equal(10))
+
+```
+
+```swift
+↓expect(10, line: 1).to(equal(10))
 
 ```
 

--- a/Source/SwiftLintFramework/Extensions/Structure+SwiftLint.swift
+++ b/Source/SwiftLintFramework/Extensions/Structure+SwiftLint.swift
@@ -26,4 +26,21 @@ extension Structure {
         parse(dictionary)
         return results
     }
+
+    internal func structures(forByteOffset byteOffset: Int) -> [[String: SourceKitRepresentable]] {
+        var results = [[String: SourceKitRepresentable]]()
+
+        func parse(_ dictionary: [String: SourceKitRepresentable]) {
+            guard let offset = dictionary.offset,
+                let byteRange = dictionary.length.map({ NSRange(location: offset, length: $0) }),
+                NSLocationInRange(byteOffset, byteRange) else {
+                    return
+            }
+
+            results.append(dictionary)
+            dictionary.substructure.forEach(parse)
+        }
+        parse(dictionary)
+        return results
+    }
 }

--- a/Source/SwiftLintFramework/Rules/Idiomatic/NimbleOperatorRule.swift
+++ b/Source/SwiftLintFramework/Rules/Idiomatic/NimbleOperatorRule.swift
@@ -21,12 +21,19 @@ public struct NimbleOperatorRule: ConfigurationProviderRule, OptInRule, Correcta
             "expect(x) === x",
             "expect(10) == 10",
             "expect(object.asyncFunction()).toEventually(equal(1))\n",
-            "expect(actual).to(haveCount(expected))\n"
+            "expect(actual).to(haveCount(expected))\n",
+            """
+            foo.method {
+                expect(value).to(equal(expectedValue), description: "Failed")
+                return Bar(value: ())
+            }
+            """
         ],
         triggeringExamples: [
             "↓expect(seagull.squawk).toNot(equal(\"Hi\"))\n",
             "↓expect(12).toNot(equal(10))\n",
             "↓expect(10).to(equal(10))\n",
+            "↓expect(10, line: 1).to(equal(10))\n",
             "↓expect(10).to(beGreaterThan(8))\n",
             "↓expect(10).to(beGreaterThanOrEqualTo(10))\n",
             "↓expect(10).to(beLessThan(11))\n",
@@ -53,7 +60,7 @@ public struct NimbleOperatorRule: ConfigurationProviderRule, OptInRule, Correcta
     fileprivate typealias Operators = (to: String?, toNot: String?)
     fileprivate typealias MatcherFunction = String
 
-    fileprivate let operatorsMapping: [MatcherFunction: Operators] = [
+    private let operatorsMapping: [MatcherFunction: Operators] = [
         "equal": (to: "==", toNot: "!="),
         "beIdenticalTo": (to: "===", toNot: "!=="),
         "beGreaterThan": (to: ">", toNot: nil),
@@ -84,6 +91,20 @@ public struct NimbleOperatorRule: ConfigurationProviderRule, OptInRule, Correcta
             .filter { _, kinds in
                 kinds.filter(excludingKinds.contains).isEmpty && kinds.first == .identifier
             }.map { $0.0 }
+            .filter { range in
+                let contents = file.contents.bridge()
+                guard let byteRange = contents.NSRangeToByteRange(start: range.location, length: range.length) else {
+                    return false
+                }
+
+                let containsCall = file.structure.structures(forByteOffset: byteRange.upperBound - 1)
+                    .contains(where: { dict -> Bool in
+                        return dict.kind.flatMap(SwiftExpressionKind.init) == .call &&
+                            (dict.name ?? "").starts(with: "expect")
+                    })
+
+                return containsCall
+            }
     }
 
     public func correct(file: File) -> [Correction] {

--- a/Source/SwiftLintFramework/Rules/Idiomatic/UnneededBreakInSwitchRule.swift
+++ b/Source/SwiftLintFramework/Rules/Idiomatic/UnneededBreakInSwitchRule.swift
@@ -87,22 +87,3 @@ public struct UnneededBreakInSwitchRule: ConfigurationProviderRule, AutomaticTes
         return patternEnds.max()
     }
 }
-
-private extension Structure {
-    func structures(forByteOffset byteOffset: Int) -> [[String: SourceKitRepresentable]] {
-        var results = [[String: SourceKitRepresentable]]()
-
-        func parse(_ dictionary: [String: SourceKitRepresentable]) {
-            guard let offset = dictionary.offset,
-                let byteRange = dictionary.length.map({ NSRange(location: offset, length: $0) }),
-                NSLocationInRange(byteOffset, byteRange) else {
-                    return
-            }
-
-            results.append(dictionary)
-            dictionary.substructure.forEach(parse)
-        }
-        parse(dictionary)
-        return results
-    }
-}

